test/src/main/java/org/apache/accumulo/test/util/Wait.java

```diff
@@ -29,23 +29,37 @@ public interface Condition {
     boolean isSatisfied() throws Exception;
   }
 
-  public static boolean waitFor(Condition condition) throws Exception {
-    return waitFor(condition, MAX_WAIT_MILLIS);
+  /**
+   * Wait for the provided condition - will throw an IllegalStateException is the wait exceeds the
+   * wait period.
+   */
+  public static void waitFor(Condition condition) throws Exception {
+    waitFor(condition, MAX_WAIT_MILLIS);
   }
 
-  public static boolean waitFor(final Condition condition, final long duration) throws Exception {
-    return waitFor(condition, duration, SLEEP_MILLIS);
+  /**
+   * Wait for the provided condition - will throw an IllegalStateException is the wait exceeds the
+   * wait period.
+   */
+  public static void waitFor(final Condition condition, final long duration) throws Exception {
+    waitFor(condition, duration, SLEEP_MILLIS);
   }
 
-  public static boolean waitFor(final Condition condition, final long duration,
-      final long sleepMillis) throws Exception {
+  /**
+   * Wait for the provided condition - will throw an IllegalStateException is the wait exceeds the
+   * wait period.
+   */
+  public static void waitFor(final Condition condition, final long duration, final long sleepMillis)
+      throws Exception {
```

Comment:
Also I think you should be able to get rid of the throws Exception here if just throwing a runtime exception only

Reply:
The exception from `condition.isSatisfied()` needs to be declared, or turned into a Runtime exception - otherwise it is bubbling up through the waitFor calls - and the InterruptedException is not being specifically handled.

Reply:
Addressed in 0031099059